### PR TITLE
Updating Canvas to handle Videos and allow tagging - AB#16580

### PR DESCRIPTION
**Pull request by tmiller-msft**
- Hook video player into canvas
- Update styles base on pause state to allow tagging of individual frames of video
- Remove video player component from AssetPreview (more simplified)

### CODE REVIEW

File: src/react/components/pages/editorPage/canvas.tsx [hunk 1]
```diff
@@ -351,38 +376,87 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
...
     private updateEditor = () => {
         this.deleteAllRegions();
+        // We need to check if we're looking for a video or image
+        if (this.props.selectedAsset.asset.type === AssetType.Image) {
+            this.loadImage();
+        } else if (this.props.selectedAsset.asset.type === AssetType.Video) {
+            this.loadVideo();
+        } else {
+            // We don't know what type of asset this is?
+            throw new Error(strings.editorPage.assetError);
```
Comment (JacopoMangiavacchi) [inline src/react/components/pages/editorPage/canvas.tsx]: Just wonder why we have an unknown asset type?

Comment (wbreza) [inline src/react/components/pages/editorPage/canvas.tsx]: FYI - we have a special `AppError` that we are throwing in other areas that handle error localization in 1 place.  Do a search for AppError and create a new ErrorCode that can then participate in the localization.

File: src/react/components/pages/editorPage/canvas.tsx [hunk 2]
```diff
@@ -351,38 +376,87 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
...
-            }
+            this.updateRegions();
         });
         image.src = this.props.selectedAsset.asset.path;
     }
 
+    private updateRegions() {
+        if (this.props.selectedAsset.regions.length) {
+            this.props.selectedAsset.regions.forEach((region: IRegion) => {
+                const loadedRegionData = new RegionData(region.boundingBox.left,
```
Comment (wbreza) [inline src/react/components/pages/editorPage/canvas.tsx]: I know this isn't your code - but would be a good idea to make a generic transform function that can covert a `IRegion` from our app model into the `RegionData` instance that canvas tool expects.

File: src/react/components/pages/editorPage/canvas.tsx (hunk 1 above)
Comment (tmiller-msft) [reply src/react/components/pages/editorPage/canvas.tsx]: Added new Canvas error code

**Pull request merged**